module-system/sov-modules-macros/src/manifest.rs

```diff
@@ -48,12 +49,16 @@ impl<'a> Manifest<'a> {
     /// The `parent` is used to report the errors to the correct span location.
     pub fn read_constants(parent: &'a Ident) -> Result<Self, syn::Error> {
         let manifest = "constants.json";
-        let initial_path = match env::var("CONSTANTS_MANIFEST") {
+        let initial_path = match env::var("CONSTANTS_MANIFEST").and_then(|v| if v.is_empty() {
+            Err(env::VarError::NotPresent)
```

Comment:
The previous implementation intended to suppress env var errors as it might not be set.

The proposed change introduces an improvement that is to check a specific variant where the key/value pair of the environment has the value empty.

I think we will still fall into the line `64` `Err(_)` pattern, as we will correctly raise the error, but then fallback to reading from manifest dir. The most consistent behavior would be to fail if the user `export CONSTANTS_MANIFEST=""`.

I think the best way to achieve that is to add a variant of the `match` pattern, before line `57`, with `Ok(p) p.is_empty() => `, and throw an error

Reply:
Yeah, I think this makes sense. Adjusting.